docs/source/reference/respawn/dialog.rst

```diff
@@ -0,0 +1,149 @@
+===================================================       =========================================================================
+
+Related structs 
+---------------
+
+DialogMessageRuiData:
+^^^^^^^^^^^^^^^^^^^^^
+
+    * string message = ""
+    * vector style1Color = <1.0, 1.0, 1.0>
+    * vector style2Color = <0.5, 0.5, 0.5>
+    * vector style3Color = <0.5, 0.5, 0.5>
+    * float style1FontScale = 1.0
+    * float style2FontScale = 1.0
+    * float style3FontScale = 1.0
+
+
+DialogButtonData
+^^^^^^^^^^^^^^^^
+
+	* string label
+	* void functionref() activateFunc
+	* string focusMessage
+	* bool startFocused
+
+
+
+DialogFooterData
+^^^^^^^^^^^^^^^^
+
+	* string label
+	* void functionref() activateFunc
+
+
+Functions
+---------
+
+**Definition:**
+
+.. cpp:function:: void OpenDialog( DialogData dialog )
+
+    Shows the local player the dialog with the data from the struct.
+
+    :param DialogData dialog: Instance of a DialogData struct
+
+**Example:**
+
+.. code-block::
+
+    DialogData dialog
+    dialog.message = "Hello there"
+    OpenDialog( dialog )
+
+**Definition:**
+
+.. cpp:function:: void AddDialogButton( DialogData dialog, string text, void functionref() callback )
+
+    Add one button to the given struct
+
+    :param DialogData dialog: Instance of a DialogData struct
+
+    :param string text: The Text that is shown on the button, supports some assets with ``%ASSET PATH%``
+
+    :param void functionref() callback: Function that is executed when the button is pressed.
+
+**Example:**
+
+.. code-block::
+
+    void function SendDialogWithButton()
+    {
+        DialogData dialog
+        dialog.message = "Hello there"
+        AddDialogButton(dialog, "Button 1 %%$r2_ui/menus/loadout_icons/primary_weapon/primary_kraber%%", void function():() {
+            printt( "pressed button 1" )
+        })
+        OpenDialog( dialog )
+    }
+
+**Definition:**
+
+.. cpp:function:: void AddDialogFooter( DialogData dialog, string text)
+
+    Adds a footer to the dialog struct
+
+    :param DialogData dialog: Instance of a DialogData struct
+
+    :param string text: The Text that is shown on the button, supports some assets with ``%ASSET PATH%``
+
+
+**Definition:**
+
+.. cpp:function:: bool IsDialogActive( DialogData dialogData )
+
+    :param DialogData dialog: Instance of a DialogData struct
+
+    :returns: ``true`` if the dialog with that struct is currently open, otherwise it returns ``false``
+
+Code example
+------------
+
+the folowing code produces this output: 
+
+.. code-block::
+
+    DialogData dialog
+    dialog.header = "This is the header"
+    dialog.message = "this is the body, it is green \n \n Hello There \n \n General Kenobi"
+    dialog.messageColor = [0,200,0,100]
+    dialog.showSpinner = true
+    dialog.showPCBackButton = true
+    AddDialogButton( dialog, "Button 1 %%$r2_ui/menus/loadout_icons/primary_weapon/primary_kraber%%", ButtonOnePressed )
+    SendServerDialog( player, dialog )
```

Comment:
if this is a thing what was even the point of server dialogs

Reply:
No im wrong, my example was copy pasted from dialog and not cleaned up enough will fix it all, when I'm home